optimism/src/keccak/mod.rs

```diff
@@ -70,6 +79,77 @@ pub enum Constraint {
     IotaStateG(usize),
 }
 
+#[allow(dead_code)]
+/// The Keccak circuit
+pub(crate) type KeccakCircuit<F> = Circuit<ZKVM_KECCAK_COLS, Steps, F>;
```

Comment:
I'm actually wondering. This thing builds both constraints and the witness, so it's kind of like a `ConstraintBuilderEnv` joined with `WitnessBuilderEnv`. I'm more pro keeping these separate, because (in the real world not tests) the verifier will need to use the former but not the latter. The real verifier does not know the witness. Or am I not seeing how this `KeccakCircuit` can be used on the verifier side?